crates/solidity/inputs/language/src/definition.rs

```diff
@@ -0,0 +1,3854 @@
+                            )
+                        ),
+                        Repeated(
+                            name = StateVariableAttributes,
+                            repeated = NonTerminal(StateVariableAttribute),
+                            allow_empty = true
+                        ),
+                        Enum(
+                            name = StateVariableAttribute,
+                            variants = [
+                                Variant(
+                                    name = Override,
+                                    fields = (specifier = Required(NonTerminal(OverrideSpecifier)))
+                                ),
+                                Variant(
+                                    name = Constant,
+                                    fields = (keyword = Required(Terminal([ConstantKeyword])))
+                                ),
+                                Variant(
+                                    name = Internal,
+                                    fields = (keyword = Required(Terminal([InternalKeyword])))
+                                ),
+                                Variant(
+                                    name = Private,
+                                    fields = (keyword = Required(Terminal([PrivateKeyword])))
+                                ),
+                                Variant(
+                                    name = Public,
+                                    fields = (keyword = Required(Terminal([PublicKeyword])))
+                                ),
+                                Variant(
+                                    name = Immutable,
+                                    enabled_in = "0.6.5",
+                                    fields = (keyword = Required(Terminal([ImmutableKeyword])))
+                                )
+                            ]
+                        )
+                    ]
+                ),
+                Topic(
+                    title = "Functions",
+                    items = [
+                        Struct(
+                            name = FunctionDefinition,
+                            fields = (
+                                function_keyword = Required(Terminal([FunctionKeyword])),
+                                name = Required(Terminal([
+                                    Identifier,
+                                    FallbackKeyword,
+                                    ReceiveKeyword
+                                ])),
+                                parameters = Required(NonTerminal(ParametersDeclaration)),
+                                attributes = Required(NonTerminal(FunctionAttributes)),
+                                returns = Optional(reference = NonTerminal(ReturnsDeclaration)),
+                                body = Required(NonTerminal(FunctionBody))
+                            )
+                        ),
+                        Struct(
```

Comment:
I prefer this to be an explicit `DelimitedBy` construct

Reply:
Discussing offline.

Reply:
Added `error_recovery` field.